src/components/panel/panel.js

```diff
@@ -806,6 +850,75 @@ MdPanelService.prototype.newPanelAnimation = function() {
 
 
 /**
+ * Creates a panel group and adds it to a tracked list of panel groups.
+ * @param groupName {string} Name of the group to create.
+ * @param config (!Object=) Specific configuration object that may contain the
+ *     following properties:
+ *
+ *   - `maxOpen` - `{number=}`: The maximum number of panels that are allowed
+ *     open within a defined panel group.
+ */
+MdPanelService.prototype.newPanelGroup = function(groupName, config) {
+  if (!this._groups[groupName]) {
+    config = config || {};
+    var group = {
+      panels: [],
+      maxOpen: config.maxOpen > 0 ? config.maxOpen : false
+    };
+    this._groups[groupName] = group;
+  }
+};
+
+
+/**
+ * Set the maximum number of panels in a group that can be opened at a given
+ * time.
+ * @param {string} groupName The name of the group to configure.
+ * @param {number|boolean} maxOpen The maximum number of panels that can be
+ *     opened. If false is passed in, the maximum limit will be removed.
+ */
+MdPanelService.prototype.setGroupMaxOpen = function(groupName, maxOpen) {
+  if (this._groups[groupName]) {
+    this._groups[groupName].maxOpen = maxOpen;
+  }
+};
+
+
+/**
+ * Determines if opening a panel within a group will exceed the limit of allowed
+ * open panels. If so, the first panel opened within the group will be closed.
+ * @param {string} groupName The name of the group to check.
+ * @private
+ */
+MdPanelService.prototype._checkGroupMaxOpen = function(groupName) {
+  if (this._groups[groupName] && this._groups[groupName].maxOpen > 0) {
+    var openCount = 0,
+        firstOpenedPanel = null,
+        firstOpenedPanelIndex = null;
+
+    angular.forEach(this._groups[groupName].panels, function(panel, index) {
+      if (panel.isOpened) {
+        openCount++;
+        if (openCount === 1 && !firstOpenedPanel) {
```

Comment:
Instead of checking against the variable that was just incremented (`openCount`), couldn't we just check that the `index === 0` to determine that it's first?